pkgs/applications/emulators/yuzu/generic.nix

```diff
@@ -111,6 +111,11 @@ in stdenv.mkDerivation {
   # making the build fail, as that path does not exist
   dontFixCmake = true;
 
+  postPatch =  ''
+    substituteInPlace     externals/Vulkan-Headers/CMakeLists.txt \
+      --replace "add_library(Vulkan::Headers ALIAS Vulkan-Headers)" ""
```

Comment:
This is a submodule checkout of the upstream Vulkan-Headers repo, right? If `YUZU_USE_EXTERNAL_VULKAN_HEADERS=ON` is set, why do we need to mess with it at all?

Reply:
Humm, the way vulkan works, one should be able to connect an application with an outdated loader/driver that does not provide a specific extension or api-level. But I imagine in compilation-time the same does not apply, since the symbol will be MIA gcc should fail.

Reply:
I'm not sure what that has to do with my comment, but it's wrong in any case. Vulkan extensions/optional features are loaded dynamically; they're not statically exported symbols at all. The linker isn't involved.

Reply:
I meant to comment the same line, and by the compile-time I meant using the symbols defined in the vulkan's headers in a newer version but not older one (that will eventually happen considering our use-case)

Reply:
Updates to the Vulkan headers do not introduce new statically exported symbols. That would make it impossible to develop software that gracefully degrades on older systems.

Reply:
Ok, that sounds ideal.

Reply:
> why do we need to mess with it at all?

When cmake loads Vulkan that module is already defined, if I don't comment that line cmake complaints because the module is being redefined.

Reply:
Originally I tried only setting the cmake flag, but it was not enough. Another option could be overriding the checked version, but that will require some regex mumbo-jumbo.

Reply:
> When cmake loads Vulkan that module is already defined, if I don't comment that line cmake complaints because the module is being redefined.

Hmm, it would be better to patch out their loading cmake modules from the vendored `externals/Vulkan-Headers` entirely.

Reply:
It's fine if I do:
```bash
    substituteInPlace     externals/CMakeLists.txt \
    --replace "add_subdirectory(Vulkan-Headers)" ""
```
instead ?

Reply:
That also works.

Reply:
LGTM, thanks!